src/server/services/buzz.service.ts

```diff
@@ -0,0 +1,86 @@
+import { TRPCError } from '@trpc/server';
+import { env } from '~/env/server.mjs';
+import {
+  GetUserBuzzAccountResponse,
+  GetUserBuzzAccountSchema,
+  GetUserBuzzTransactionsSchema,
+  GetUserBuzzTransactionsResponse,
+  CreateBuzzTransactionInput,
+} from '~/server/schema/buzz.schema';
+import { throwBadRequestError } from '~/server/utils/errorHandling';
+import { QS } from '~/utils/qs';
+
+export async function getUserBuzzAccount({ accountId }: GetUserBuzzAccountSchema) {
+  const response = await fetch(`${env.BUZZ_ENDPOINT}/account/${accountId}`);
+  if (!response.ok) {
+    switch (response.status) {
+      case 400:
+        throw throwBadRequestError();
+      case 404:
+        throw new TRPCError({ code: 'NOT_FOUND', message: 'Account not found' });
+      default:
+        throw new TRPCError({
+          code: 'INTERNAL_SERVER_ERROR',
+          message: 'An unexpected error ocurred, please try again later',
+        });
+    }
+  }
+
+  const data: GetUserBuzzAccountResponse = await response.json();
+  return data;
+}
+
+export async function getUserBuzzTransactions({
+  accountId,
+  ...query
+}: GetUserBuzzTransactionsSchema & { accountId: number }) {
+  const queryString = QS.stringify(query);
+  const response = await fetch(
```

Comment:
Should probably confirm the user has permissions to do this - i.e, targetAccount === sessionAccount || isMod

Reply:
We are doing it at the controller level. Leaving the service as is would allow us to reuse any other place if needed